### ISSUE TITLE
Port site to hugo

### ISSUE THREAD

**Issue by matiasandina** [open-neuroscience/open-neuroscience.github.io#10]
**Requirements**

- [x] Search Bar
- [x] Flat section for categories as in "Our works" from https://themes.gohugo.io/theme/meghna-hugo/
- [x] Multi-lingual support see #4 

Likely going to try academic theme, it comes with most of the things built-in and is easier to make it custom.

**Comment by matiasandina** [comment]
I have used the academic theme in Hugo because I thought it had most requirements from, best documentation and it's highly flexible.
I have created a temporary site on netlify so that everyone can see the deploy.

https://openneuro.netlify.com

There are a lot of things to add to this site and there are small differences to port .
What's the first impression?
I still need to add @GHPdev I couldn't find a picture.
If you like it, let's talk about how to port all the site (there are some issues here and there).

**Comment by amchagas** [comment]
Hi Matias,
nice! thanks for putting this together. Looks good! 

from what you write I have the impression that there is no way to batch process all the current posts/pages to the new system? If I'm wrong, this would make things faster. 

If we have to do everything manually, it might be worth to deploy what you have done to github, using a "sub page"? This way we can work on things in parallel, and try to move things along a bit faster.

I've created another repo called staging, where we can deploy the new website/test things and once done, we can make it the main page. https://open-neuroscience.com/staging

If you think this is good, I can do the initial deployment if you want or you could check the documentation to see how to do it yourself? (either way, just let me know if you think this is good for now, and what is the best way to deploy things)

about #6  there are a couple more people to add, that are not on the github group, but I add them later.

**Comment by matiasandina** [comment]
Great, we can probably do it in a first step automatically and then curate manually. Post files have to go in a folder with the post name, I would recommend no spaces on folder names. Posts inside are named as `index.md` and their featured picture, if any, as `featured.jpg|png`.

I'm not sure how to code this but a script that moves files and renames them can be a good start. The main idea would be to grab all `post-name.md` posts and move them to the `/content/language/post/post-name/index.md`. A good idea would be adding whatever category posts belong to as the first tag. This way we can filter them in the first section on the page.

After that, there are some things that might break.

For example, the yaml headers are not exactly the same. Notice "author**s**" and the bracket structure. Authors names should match whatever given name appears in their content/author. Multiple authors can be separated with commas.

Hierarchical organization follows the folder structure on the original site in the order of the tags. More tags might be needed for good indexing.
Right now, all the site cares about are tags, I couldn't figure out the categories but kept it just in case.
---
title: DremelFuge
authors: ["André Maia Chagas"]
layout: post
categories: [Benchtop]
tags: ["Benchtop", "centrifuges", "wet lab"]
---

I notice you had instances where you use `<figure>` to put a figure with a caption or an `<iframe>`. This doesn't work, so the curation step includes moving the <img src> part of it outside the `<figure>` or `<iframe>` and checking that it renders properly.

There are other parts of the posts that I have not yet fully figured out, like social icons at the end of the posts and stuff like that.

To render the site, I am using [`blogdown`](https://bookdown.org/yihui/blogdown/), which is what I'm familiar with and deals with building the public library in just one R command:

```
blogdown::serve_site()
``` 

I have the version control of the site here:
https://github.com/matiasandina/openneuro

I think you should be able to move this to the other repo. After that, we all start commiting there instead?

**Comment by amchagas** [comment]
Hi Matias, 
I took a bit of time today to create a python script to convert things...
right now, it only creates the necessary file structure and changes the layout in the md file from page to post. 

I think all folder names don't have spaces in them. There is just a mixed usage of hyphen and underscore, but that should be easy to programmatically change as well.

All changes and the script (jupyter notebook) were dumped here: https://github.com/open-neuroscience/jekyll2hugo

Let me know if this works for the current implementation, or which errors it spits out?

**Comment by amchagas** [comment]
It is a bit funny that there are errors with those html elements, as I thought Markdown was able to deal with them... 

Maybe something specific to the Hugo theme. The reason I used them is that it was the only way I found at the time to have them properly displayed on the respective pages.

**Comment by matiasandina** [comment]
So,
The script was quite there.  
A couple of errors but I modified and was able to port all content into something close enough to an actual post.
I think the things that remain are more difficult to implement automatically. 
Site is online at openneuro.netlify.com

**Comment by amchagas** [comment]
cool, 
maybe it is a good time to port this github then? This way we can make a list of manual things needed and start working on them in parallel?

**Comment by matiasandina** [comment]
Sure, can it be done by uploading all folders to https://github.com/open-neuroscience/staging ?

**Comment by amchagas** [comment]
hi, more or less. I've staged everything so that we can start deploying things directly from github:

We should work on this repo: https://github.com/open-neuroscience/academic-kickstart
There, the hugo theme is "installed" and running "hugo" creates content in the "public" folder in this repo.

The folder "public" is actually a submodule pointing to the staging repository. So once the content of this folder is changed (using the step above), you can go into the folder and push those changes directly to the remote repository on github. This should automatically update open-neuroscience.com/staging

(hope this is clear enough?)

Like this we can have all changes tracked and open in the repos...

**Comment by amchagas** [comment]
I've basically made small changes to the instructions here https://sourcethemes.com/academic/docs/deployment/ (on the github pages section

**Comment by matiasandina** [comment]
I'm not really sure, I have seriously modified config files and folder structure. 
I have already run the "hugo" command (`blogdown::serve_site()` ends with building it). The folder structure that I have is the result of building the site.

If I understand correctly we need to replace,

`config.toml`
all config files in `_config` folder
All content in the `content` folder

Besides, we would need `config.toml` to point to the new URL as the base URL.
I can try this PR and we can work back from there.

**Comment by amchagas** [comment]
I've changed the config.toml file at the root of the repository, and got the site to be deployed here   https://open-neuroscience.com/staging/en/
I think we are good to start making the other necessary changes...
Do you have a list of things that need to be changed at this point?

**Comment by matiasandina** [comment]
GREAT!

Here's a list of what I can think of right now. Some of them might be better moved as issues, so that we close this one and move on? 

- [ ] Try functionality of the contact section
- [ ] Current logo doesn't fit well on the circle, reshape ?
- [ ] fix favicon
- [ ] find posts with `iframe` or `figure` and fix if necessary
- [ ] There are posts that used to be "categories" on the old site [example](https://open-neuroscience.com/staging/en/post/hardware/). Fix/remove them, they are showing code to list posts within categories.
- [ ] Decide what social icons 
- [ ] Complete the author pages
- [ ] Posts without date will show as January 1st 0001, either assign correct/current date or remove date
- [ ] The topics part must include all topics, to fix that:

We should include more options on this file

https://github.com/open-neuroscience/academic-kickstart/blob/master/content/en/home/topics.md

The syntax goes like:

```
  [[content.filter_button]]
    name = "Benchtop"
    tag = "Benchtop"
```

- [ ] Come up with a more standarized post structure or will we do this just for future posts?
- [ ] Also should we have something that says "this project is no longer maintained/has moved to..." ?
- [ ] Feature enhancement but, if seems reasonable, find a way to include a second search bar at the very top on the home page, as a new widget on top of "Topics" section
- [ ] Is the wordcloud of the usage of terms accurate/looks good/functional? We can modify it or delete it as needed. The file that handles that is https://github.com/open-neuroscience/academic-kickstart/blob/master/content/en/home/tags.md

**Comment by amchagas** [comment]
Hi,
I agree with creating issues for some of these. Maybe good to do it in the academic kickstart repo? https://github.com/open-neuroscience/academic-kickstart.


- I would remove the posts that are just for categories (the ones that are just showing code) maybe using them as guide for topics for now?
- I can add the social media addresses that are existing for Open Neurosciences (facebook and twitter), and remove the other ones. 
- The example for the broken link post is something that needs minor corrections I think, since it is something I created  a long time ago.. For other links it would be nice to figure something out. (I guess leave it as "nice to have"?

Will try to work on some of these later!

**Comment by matiasandina** [comment]
Ok, I think you need to allow for us to make issues on that one because it's a fork, see here:

https://softwareengineering.stackexchange.com/questions/179468/forking-a-repo-on-github-but-allowing-new-issues-on-the-fork

**Comment by amchagas** [comment]
Done!

**Comment by matiasandina** [comment]
Ok I'm closing this one because I just made all of the previous their own issues and we can work individually on those.
